arangod/RocksDBEngine/ReplicatedRocksDBTransactionCollection.cpp

```diff
@@ -120,7 +120,8 @@ Result ReplicatedRocksDBTransactionCollection::abortTransaction() {
 }
 
 void ReplicatedRocksDBTransactionCollection::beginQuery(
-    ResourceMonitor* resourceMonitor, bool isModificationQuery) {
+    std::shared_ptr<ResourceMonitor> resourceMonitor,
+    bool isModificationQuery) {
   auto* trxMethods = dynamic_cast<RocksDBTrxMethods*>(_rocksMethods.get());
   if (trxMethods) {
     trxMethods->beginQuery(resourceMonitor, isModificationQuery);
```

Comment:
nit: resource monitor could be moved here.
```suggestion
    trxMethods->beginQuery(std::move(resourceMonitor), isModificationQuery);
```